modules/queue/logic/queueLogic.js

```diff
@@ -0,0 +1,106 @@
+const { filter } = require('rxjs/operators');
+
+const RedisSMQ = require('rsmq');
+const { Subject } = require('rxjs');
+const logger = require('../../../utils/logger')(module);
+const { MESSAGE_QUEUES, MESSAGES } = require('../constants/queue');
+
+const rsmq = new RedisSMQ({ host: process.env.REDIS_HOST, port: process.env.REDIS_PORT || 6379, ns: 'rsmq' });
+
+// TODO TEST
+class MessageQueue {
+  queues = [];
+
+  constructor() {
+    // Setup all queues
+    this.interval = setInterval(() => this.notifySubscriber(), 100);
```

Comment:
The good thing there is a read timeout, so basically when the message gets retrieved from RSMQ it will not be surfaced again for X amount of time. Currently I think we have 30 seconds. If the message gets deleted in the meantime it is dealt with, otherwise the message will be sent to the subscribers again. That way the system will also scale with multiple workers, if they have their own queue listener. This system will have to be experiemented with though, when the time comes.

Reply:
So, are you using polling instead of subscription to make it compatible with several workers?

Reply:
Yes but it looks like I understood the docs incorrectly, so the update will be issued for a single queue and each subscriber can then call receiveMessage. This will be a bit of a race condition but we can still limit this to a single instance if we need to.